vllm/entrypoints/llm.py

```diff
@@ -251,4 +256,23 @@ def _run_engine(self, use_tqdm: bool) -> List[RequestOutput]:
         # This is necessary because some requests may be finished earlier than
         # its previous requests.
         outputs = sorted(outputs, key=lambda x: int(x.request_id))
-        return outputs
\ No newline at end of file
+        return outputs
+
+    def _validate_prompt(
+        self,
+        prompt: Optional[str] = None,
+        prompt_token_ids: Optional[List[int]] = None,
+        truncate_prompt_tokens: Optional[Annotated[int, Field(ge=1)]] = None
+    ) -> Tuple[Optional[str], Optional[List[int]]]:
+
+        if not (prompt or prompt_token_ids):
+            raise ValueError("Either prompt or prompt_ids should be provided.")
+        if (prompt and prompt_token_ids):
+            raise ValueError(
+                "Only one of prompt or prompt_ids should be provided.")
+        if truncate_prompt_tokens is not None:
+            if prompt_token_ids is None:
+                prompt_token_ids = self.llm_engine.tokenizer.tokenizer(
+                    prompt).input_ids
+            prompt_token_ids = prompt_token_ids[-truncate_prompt_tokens:]
```

Comment:
This isn't exactly the same behaviour as the [equivalent code](https://github.com/vllm-project/vllm/blob/0650e5935b0f6af35fb2acf71769982c47b804d7/vllm/entrypoints/openai/serving_engine.py#L201-L210) in the completion API. In that case, we use the built-in capability of the tokenizer to perform truncation in the case that the user provides a prompt, and only implement the truncation ourselves in the case when the user provides prompt ids. I think it would be better to do the same here to be consistent.

Reply:
yes indeed, the thing I remarked though is that using the tokenizer with kwargs does a right truncation and not a left one basically doing prompt_token_ids[:truncated_prompt_tokens]. I ll recheck to make sure I am not mistaken.

Reply:
regarding the build I got an error `buildkite/ci/pr/distributed-tests-multiple-groups — Failed (exit status -10)`  do you have any idea what could it be ? and how could I debug this ?

Reply:
@yecohn Regarding the truncation side issue. You are right, that by default the tokenizers do truncation from the right. That's why we added[ this argument ](https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/openai/serving_engine.py#L77) when calling `get_tokenizer` to set the truncation side to left. 

Re: the build error I don't think it looks related to your changes currently.